flutter/lib/src/event_processor/widget_event_processor.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+
+import '../../sentry_flutter.dart';
+
+class WidgetEventProcessor implements EventProcessor {
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) {
+    if (event is SentryTransaction) {
+      return event;
+    }
+    if (event.exceptions == null && event.throwable == null) {
+      return event;
+    }
+    final context = sentryWidgetGlobalKey.currentContext;
+    if (context == null) {
+      return event;
+    }
+    final textScaleFactor =
+        MediaQuery.maybeTextScalerOf(context)?.textScaleFactor;
+    if (textScaleFactor == null) {
+      return event;
+    }
+    final textScale =
+        textScaleFactor == 1.0 ? 'no scaling' : 'linear (${textScaleFactor}x)';
```

Comment:
You can just use `textScaler?.toString() ?? 'no-scaling' here and avoid the deprecation.

Reply:
Looks like `toString` is only overridden in `_LinearTextScaler` class, while `_ClampedTextScaler` does not override `toString`.

Reply:
Right... but since you can't figure out if it is clamped anyway, because of private classes, maybe just go with `textScaler?.toString() ?? 'unknown'`?

Reply:
Also I don't know in which Flutter version `TextScaler` was introduced. Need to be careful about not breaking backwards compatibility.

Reply:
Don't really know that the output of `toString` will be when it's non-null. Opted to use the `scale(fontSize)` function for now with value `1`. Does this add enough context so that it's useful? Don't really know how else we should handle non-liniear scalers.

Reply:
@kuhnroyal Any additional thoughts?

Reply:
The `TextScaler` was only added in Flutter 3.16 - see https://docs.flutter.dev/release/breaking-changes/deprecate-textscalefactor?
So there is probably a switch needed, or for now just live with the deprecation and use the `textScaleFactor`.

Reply:
@denrase can we use the dynamic approach here that we do for other code that needs backwards compatibility?

Reply:
If we only have this static method to access the text scaler, we cannot use the dynamic approach. In that case we can only use the deprecated method available on all flutter versions we have to support.